Sources/TuistGenerator/Generator/LinkGenerator.swift

```diff
@@ -196,7 +196,10 @@ final class LinkGenerator: LinkGenerating {
         if precompiledFrameworkPaths.isEmpty {
             precompiledEmbedPhase.shellScript = "echo \"Skipping, nothing to be embedded.\""
         } else {
-            let script = try embedScriptGenerator.script(sourceRootPath: sourceRootPath, frameworkPaths: precompiledFrameworkPaths)
+            let isTestTarget = [.ocUnitTestBundle, .uiTestBundle, .unitTestBundle].contains(pbxTarget.productType)
```

Comment:
You can leverage `target.product.testsBundle` which already captures this logic :)

May have to update the method signature to accept `target: Target` (one of the tuist model types).

Reply:
Do you know if there is a function from `(PBXTarget) -> Target` (a swift version of Hoogle would be awesome for finding this kind of thing out).

Reply:
There isn't a need to map `PBXTarget` to `Target` we can pass it from the calling function where `Target` is available.

https://github.com/tuist/tuist/pull/1118/files/5a4e25ac214a40f615aa16d83f8640d922396536#diff-bd14a50b734b61af028ddef07944c580R74

https://github.com/tuist/tuist/pull/1118/files/5a4e25ac214a40f615aa16d83f8640d922396536#diff-bd14a50b734b61af028ddef07944c580R91